docs/community/rust-coding-guidelines/rust-secure-coding-guidelines.md

```diff
@@ -0,0 +1,253 @@
+- When vetting a new library, it's advisable to utilize deps.dev for evaluation. This site provides an OpenSSF scorecard containing essential information. As a guideline, libraries with an OpenSSF score of 7 or higher are typically safe to import. However, those scoring **below 7** must undergo discussion prior to importation.
+
+### Minimize Use of Feature Flags
+
+As a general practice, avoid using feature flags in your crates unless absolutely necessary. Feature flags can introduce complexity and unexpected behaviours, making the codebase harder to audit for security vulnerabilities.
+
+### Understanding Feature Unification
+
+Be aware of Cargo's feature unification process. When multiple dependencies require the same crate with different feature flags, Cargo unifies these into a single configuration. This unification can inadvertently enable features that might not be desirable or secure for the project [[Rustbook: features unification]](https://doc.rust-lang.org/cargo/reference/features.html#feature-unification) [[Rustbook: feature resolver]](https://doc.rust-lang.org/cargo/reference/features.html#feature-resolver-version-2).
+
+## Language Generalities
+
+### Unsafe Code
+
+Never use `unsafe` blocks unless as a last resort. Justify their use in a comment, detailing how the code is effectively safe to deploy [[unsafe code]](https://anssi-fr.github.io/rust-guide/04_language.html#unsafe-code).
+
+### Integer Overflows
+
+Refer to [coding-guidelines](./rust-coding-guidelines.md#integer-arithmetic).
+
+### Error Handling
+
+Use `Result<T, E>` and `Option<T>` for error handling instead of _unwrapping_ or _expecting_, to avoid panics [[error handling]](https://anssi-fr.github.io/rust-guide/04_language.html#error-handling) [[panic]](https://anssi-fr.github.io/rust-guide/04_language.html#panics) [[coding-style]](rust-coding-guidelines.md/#error-handling).
+
+### Assertions
+
+Prefer using `Result` and context-rich error handling over Rust's `assert!`, `assert_eq!`, and `assert_ne!` macros for enforcing invariants, reserving assertions for development and unrecoverable error scenarios.
+
+## Types Systems and Data Structures
+
+### Drop Trait
+
+Implement the `Drop` trait selectively and ensure it doesn't cause panics [[drop trait the destructor]](https://anssi-fr.github.io/rust-guide/06_typesystem.html#drop-trait-the-destructor).
+
+- Implement the Drop trait selectively, only when necessary for specific destructor logic. It's mainly used for managing external resources or memory in structures like Box or Rc, often involving unsafe code and security-critical operations.
+- In a Rust secure development, the implementation of the `std::ops::Drop` trait
+  must not panic.
+- Do not rely on `Drop` trait in security material treatment after the use, use [zeroize](https://docs.rs/zeroize/latest/zeroize/#) to explicit destroy security material, e.g. private keys.
+
+### Send and Sync Traits
```

Comment:
how much is this relevant for the existing codebase?

Reply:
We have some instances of Sync + Send trait implementation.